inference-engine/src/vpu/graph_transformer/src/stages/rnn.cpp

```diff
@@ -130,14 +132,31 @@ void FrontEnd::parseRNN(const Model& model, const ie::CNNLayerPtr& _layer, const
 
     size_t nCells = inputs[0]->desc().dim(Dim::H);
     size_t nBatches = inputs[0]->desc().dim(Dim::C);
+    size_t inputSize = inputs[0]->desc().dim(Dim::W);
+
+    bool permute_input = (1 != layer->axis);
+
+    if (!permute_input) {
```

Comment:
make this condition more simple: process each `axis` case or just permute_input/!permute_input (like in GPU plugin). Please do not mix them together.